model/PermissionsService.php

```diff
@@ -163,8 +161,35 @@ private function dryAdd(array $add, core_kernel_classes_Resource $resource, arra
         }
     }
 
-    private function getResourcesToUpdate(core_kernel_classes_Resource $resource, bool $isRecursive): array
+    /**
+     * It is likely the strategy has not checked for duplicate permissions;
+     * therefore, we need an additional step to remove them.
+     *
+     * A duplicate ACL may surface in cases the same grant exists for a user
+     * both at the class and item level, and is needed here in case the strategy
+     * used doesn't handle them on its own.
+     */
+    private function deduplicateActions(array $actions): array
     {
+        foreach ($actions['add'] as &$entry) {
+            foreach ($entry['permissions'] as $_uid => &$grants) {
```

Comment:
What is `$_uid`? Can we have a better variable name?


```suggestion
            foreach ($entry['permissions'] as $_uid => &$grants) {
```

Reply:
It is the User Id. I've removed it since it was not used inside the loop: ee1ae41942e99d6bcb5572b0b5d2084b178b0a77